addons/account/models/account_invoice.py

```diff
@@ -1334,7 +1335,16 @@ def _check_duplicate_supplier_reference(self):
             # refuse to validate a vendor bill/credit note if there already exists one with the same reference for the same partner,
             # because it's probably a double encoding of the same bill/credit note
             if invoice.type in ('in_invoice', 'in_refund') and invoice.reference:
-                if self.search([('type', '=', invoice.type), ('reference', '=', invoice.reference), ('company_id', '=', invoice.company_id.id), ('commercial_partner_id', '=', invoice.commercial_partner_id.id), ('id', '!=', invoice.id)]):
+                year = fields.Date.from_string(invoice.date_invoice).year
+                date_start = datetime.date(year, 1, 1)
+                date_end = datetime.date(year, 12, 31)
```

Comment:
```suggestion
                date_start = fields.Date.from_string(invoice.date_invoice).replace(
                    day=1, month=1
                )
                date_end = fields.Date.from_string(invoice.date_invoice).replace(
                    day=31, month=12
                )
```

Reply:
As of my previous comment, here month could be removed.

Reply:
ok